### DISCUSSION TITLE
Hide Console Launcher stack trace in case of failures

### DISCUSSION THREAD

**Discussion by scordio** [junit-team/junit-framework#3022] [Q&A]
We have a Console Launcher based application that executes several tests against a remote device and checks its configuration parameters, failing with assertion errors if any of these parameters do not match the expected values.

The tree output of the console launcher provides already all the needed information to analyze failures, i.e., the ❌ next to the test name and the assertion error message. However, each test failure triggers also the printing of a stack trace, appended after the tree output.

Is there a way to hide the stack trace and keep only the tree output? I couldn't find any viable options at https://junit.org/junit5/docs/current/user-guide/#running-tests-console-launcher-options but maybe I overlooked something.

**Comment by Raphasha27** [comment]
You can hide the stack trace in Console Launcher with `--details=none`:

```bash
java -jar junit-platform-console-standalone.jar \
  --details=none \
  --select-class com.example.MyTest
```

For programmatic usage:

```java
import org.junit.platform.console.ConsoleLauncher;
import org.junit.platform.console.options.CommandLineOptions;

public class CustomRunner {
    public static void main(String[] args) {
        CommandLineOptions options = new CommandLineOptions();
        options.setDetails(Details.NONE);  // hides stack traces
        // or Details.SUMMARY for just test counts
        ConsoleLauncher.execute(options);
    }
}
```

If you have failures and want to see **just the failure message** without the full stack trace:

```bash
--details=summary  # shows pass/fail counts + failure messages only
```

Available `--details` levels:
- `none` — only warnings and errors
- `summary` — test counts + failure messages (no stack traces)
- `flat` — test names + failures (no stack traces per test)
- `tree` — hierarchical with stack traces (default)
- `verbose` — full output with stack traces

For Maven Surefire, hide stack traces in `pom.xml`:

```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-surefire-plugin</artifactId>
    <configuration>
        <trimStackTrace>false</trimStackTrace>
    </configuration>
</plugin>
```

Set `trimStackTrace` to `true` to show only the first line of each stack trace.

**Reply by sormuras** [reply]
No.

**Comment by sormuras** [comment]
> Is there a way to hide the stack trace and keep only the tree output?

No, there's none.

A tweak to `TreePrinter::printThrowable` would be required to achieve that. Something in the line of:

```java
	private void printThrowable(String indent, TestExecutionResult result) {
		if (result.getThrowable().isEmpty() || _HIDE_STACKTRACE_) {
			return;
		}
		...
	}
```
